### ISSUE TITLE
B::C crashes when testing issue #203

### ISSUE THREAD

**Issue by toddr** [CpanelInc/perl-compiler#27]
```
$>perlcc  -e 'use Data::Dumper; use open(IN => ":crlf", OUT => ":encoding(cp1252)"); open F, "<", "/dev/null"; print Dumper [ PerlIO::get_layers(F, input  => 1) ];'
/usr/local/cpanel/3rdparty/perl/522-debug/bin/perlcc: Unexpected compiler output
Can't locate object method "NAME" via package "B::CV" at /usr/local/cpanel/3rdparty/perl/522-debug/lib/perl5/cpanel_lib/i386-linux-debug-64int/B/C/OverLoad/B/PVMG.pm line 155.
 CHECK failed--call queue aborted.
/usr/lib/gcc/i686-redhat-linux/4.4.7/../../../crt1.o: In function `_start':
(.text+0x18): undefined reference to `m
```

**Comment by toddr** [comment]
I've got it.

**Comment by toddr** [comment]
This fixes things:

``` diff
diff --git a/lib/B/C/OverLoad/B/PVMG.pm b/lib/B/C/OverLoad/B/PVMG.pm
index 9bca164..e4776df 100644
--- a/lib/B/C/OverLoad/B/PVMG.pm
+++ b/lib/B/C/OverLoad/B/PVMG.pm
@@ -152,7 +152,7 @@ sub save_magic {
     if ( $sv_flags & SVf_AMAGIC ) {
         my $name = $fullname;
         $name =~ s/^%(.*)::$/$1/;
-        $name = $pkg->NAME if $pkg and $$pkg;
+        $name = $pkg->NAME if $pkg and ref $pkg and $pkg->can('NAME');
         debug( [qw/mg gv/], "initialize overload cache for %s", $fullname );

         init1()->add( sprintf( "Gv_AMG(%s); /* init overload cache for %s */", savestashpv($name), $fullname ) );
```

**Comment by toddr** [comment]
However it's not clear why we only need this on cm522. Possibly a result of our %Exporter:: logic.

**Comment by toddr** [comment]
This might be related to overload:

 [WARNING] ===== B::CV
 [WARNING] NAME became: IO::File
 [level=gv,mg] initialize overload cache for %IO::File::

**Comment by toddr** [comment]
fixed in a55154e8ddc9002f368cedac4f838d7b15c85455
